packages/firebase_messaging/android/src/main/java/io/flutter/plugins/firebasemessaging/FirebaseMessagingPlugin.java

```diff
@@ -60,7 +60,16 @@ public void onReceive(Context context, Intent intent) {
     } else if (action.equals(FlutterFirebaseMessagingService.ACTION_REMOTE_MESSAGE)) {
       RemoteMessage message =
           intent.getParcelableExtra(FlutterFirebaseMessagingService.EXTRA_REMOTE_MESSAGE);
-      channel.invokeMethod("onMessage", message.getData());
+
+      Map<String, Object> content = new HashMap<>();
+      content.put("data", message.getData());
+
+      RemoteMessage.Notification notification = message.getNotification();
```

Comment:
Since notification can be null if the a data-message is sent, please check for null before adding the properties of the Notification to the content passed to the Dart side.